nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -0,0 +1,142 @@
+package com.uber.nullaway;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Attribute;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.HashSet;
+import java.util.List;
+
+public class GenericsChecks {
+
+  // check that type is a valid instantiation of its generic type
+  public static void checkInstantiatedType(
+      Type type, VisitorState state, Tree tree, NullAway analysis, Config config) {
+    CodeAnnotationInfo codeAnnotationInfo = CodeAnnotationInfo.instance(state.context);
+    if (config.isJSpecifyMode()) {
```

Comment:
Rather than indenting the whole method, let's just write `if (!config.isJSpecifyMode()) { return; }` at the beginning of the function